### PR TITLE
Add unknown property warning for use of `autofocus`

### LINKED ISSUE #3248: Autofocus should warn like for attribute.
If user writes for instead of htmlFor, warning is shown. I suppose autofocus instead of autoFocus should warn too.

### PR DESCRIPTION
**Pull request by hkal**
Resolves #3248 
Resubmission of #6461

cc @Aweary @zpao

### CODE REVIEW

File: src/renderers/dom/shared/DOMProperty.js [hunk 1]
```diff
@@ -193,9 +193,13 @@ var DOMProperty = {
...
    * Mapping from lowercase property names to the properly cased version, used
    * to warn in the case of missing properties. Available only in __DEV__.
+   *
+   * autofocus is predefined, because adding it to the property whitelist
+   * causes unintended side effects.
+   *
    * @type {Object}
    */
-  getPossibleStandardName: __DEV__ ? {} : null,
+  getPossibleStandardName: __DEV__ ? {autofocus: 'autoFocus'} : null,
```
Comment (Aweary) [inline src/renderers/dom/shared/DOMProperty.js]: Can you add a space before the key name and after the value? `{ autofocus: 'autoFocus' }`

**Comment by Aweary** [comment]
Accepted pending the small style change, I'll leave this open for @zpao just in case 😄

File: src/renderers/dom/shared/DOMProperty.js (hunk 1 above)
Comment (hkal) [inline src/renderers/dom/shared/DOMProperty.js]: Done.

File: src/renderers/dom/shared/DOMProperty.js [hunk 2]
```diff
@@ -193,9 +193,13 @@ var DOMProperty = {
...
    * Mapping from lowercase property names to the properly cased version, used
    * to warn in the case of missing properties. Available only in __DEV__.
+   *
+   * autofocus is predefined, because adding it to the property whitelist
+   * causes unintended side effects.
+   *
    * @type {Object}
    */
-  getPossibleStandardName: __DEV__ ? {} : null,
+  getPossibleStandardName: __DEV__ ? { autofocus: 'autoFocus' } : null,
```
Comment (Aweary) [inline src/renderers/dom/shared/DOMProperty.js]: I'm so sorry @hkal, I guess we don't pad inline objects 😭 . You were totally right the first time.

Comment (hkal) [inline src/renderers/dom/shared/DOMProperty.js]: No worries @Aweary. I'll change it back.

**Pull request merged**